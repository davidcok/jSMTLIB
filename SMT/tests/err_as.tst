; tests illegal uses of as identifiers
(set-logic AUFLIRA)
(assert (= ((as + Int) 4 3) 1)) ;; unneeded disambiguation
(assert (= ((as and Bool) true false) false)) ;; not overloaded
(assert (= ((as zzz Bool) true false) false)) ;; unknown identifier
(declare-sort S 0)
(assert (= ((as + S) 4 3) 1))  ;; not overloaded to S
