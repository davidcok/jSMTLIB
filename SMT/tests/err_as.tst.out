success
(assert (= ((as + Int) 4 3) 1)) ;; unneeded disambiguation
           ^^^^^^^^^^^^^^^^
(error "Unknown predicate symbol + with argument types Int Int")
(assert (= ((as and Bool) true false) false)) ;; not overloaded
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
(error "Unknown predicate symbol and with argument types Bool Bool")
(assert (= ((as zzz Bool) true false) false)) ;; unknown identifier
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
(error "Unknown predicate symbol zzz with argument types Bool Bool")
success
(assert (= ((as + S) 4 3) 1))  ;; not overloaded to S
           ^^^^^^^^^^^^^^
(error "Unknown predicate symbol + with argument types Int Int")
